internal/runner/jest.go

```diff
@@ -0,0 +1,86 @@
+package runner
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"slices"
+
+	"github.com/buildkite/test-splitter/internal/debug"
+	"github.com/buildkite/test-splitter/internal/plan"
+	"github.com/kballard/go-shellquote"
+)
+
+type Jest struct {
+	TestCommand            string
+	TestFileExcludePattern string
+	TestFilePattern        string
+	RetryTestCommand       string
+}
+
+func NewJest(j Jest) *Jest {
+	if j.TestCommand == "" {
+		j.TestCommand = "yarn test -- {{testExamples}}"
+	}
+
+	if j.TestFilePattern == "" {
+		j.TestFilePattern = "**/{__tests__/**/*,*.spec,*.test}.{ts,js,tsx,jsx}"
+	}
+
+	return &j
+}
+
+// GetFiles returns an array of file names using the discovery pattern.
+func (j Jest) GetFiles() ([]string, error) {
+	debug.Println("Discovering test files with include pattern:", j.TestFilePattern, "exclude pattern:", j.TestFileExcludePattern)
+	files, err := discoverTestFiles(j.TestFilePattern, j.TestFileExcludePattern)
+	debug.Println("Discovered", len(files), "files")
+
+	if err != nil {
+		return nil, err
+	}
+
+	if len(files) == 0 {
+		return nil, fmt.Errorf("no files found with pattern %q and exclude pattern %q", j.TestFilePattern, j.TestFileExcludePattern)
+	}
+
+	return files, nil
+}
+
+func (j Jest) Command(testCases []string) (*exec.Cmd, error) {
```

Comment:
I'm thinking of removing `Command` and `RetryCommand` and introduce `Run` to the runner interface that will execute the command, as well as parsing the result.
That will also enable us to unit test the command execution.
I have a draft [PR](https://github.com/buildkite/test-splitter/pull/156) for reference.